### PR TITLE
Add mouse support to ratatui dashboard (issue #734)

### PR DESCRIPTION
**Pull request by madmax983**
## Summary
Adds comprehensive mouse support to the `--ui ratatui` dashboard, enabling scroll-wheel navigation and click-to-select functionality while maintaining keyboard-only usability.

## Key Changes

- **Mouse event handling**: Implemented `handle_mouse()` function that processes scroll wheel and left-click events, mirroring keyboard navigation behavior
  - Scroll wheel moves selection in step feed (3 rows per notch by default) or scrolls detail inspector when open
  - Left-click selects visible step rows in the feed
  - Mouse input is blocked when modals own focus (confirm prompt, help overlay, stop confirmation, search)

- **Terminal mouse capture**: Added `EnableMouseCapture` on startup and `DisableMouseCapture` on teardown to enable SGR mouse mode, with graceful fallback for unsupported terminals

- **State tracking**: Extended `DashboardState` with:
  - `feed_top_row`: Captures the absolute terminal row of the first visible feed line each frame, enabling click-to-row translation without duplicating layout math
  - `mouse_scroll_step`: Configurable rows moved per scroll-wheel notch (default 3)

- **Configuration**: Added `MAXWELL_MOUSE_SCROLL_STEP` environment variable support for customizing scroll sensitivity, with fallback to default of 3 rows

- **Documentation**: Updated README with mouse usage instructions and added mouse bindings to the help overlay keybinding reference

- **Testing**: Added 18 comprehensive tests covering scroll wheel behavior, click selection, boundary conditions, modal focus blocking, and environment variable configuration

## Implementation Details

- Mouse events are processed in the same event loop as keyboard events, with identical focus rules to prevent modal interference
- Click translation uses captured `feed_top_row` and `viewport_height` to map screen coordinates to log indices
- Scroll wheel behavior differs by context: moves cursor in step feed, scrolls in monitor mode, and scrolls detail inspector when open
- All mouse capture operations are best-effort (like bracketed paste) to maintain compatibility with terminals that don't support SGR mouse mode

https://claude.ai/code/session_01ERpfLg7hRYqPWi3Gcf5heh

### CODE REVIEW

File: src/agent/confirm_tui.rs [hunk 1]
```diff
@@ -1730,6 +1768,79 @@ fn handle_key(dash: &Arc<RatatuiDashboard>, key: KeyEvent) {
...
+                if s.detail_open {
+                    let max_scroll = get_max_detail_scroll(&s);
+                    s.detail_scroll_top = (s.detail_scroll_top + 1).min(max_scroll);
+                } else if s.is_monitor {
+                    perform_scroll(&mut s, KeyCode::Down);
+                } else {
+                    move_cursor_down(&mut s);
+                }
+            }
+        }
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/agent/confirm_tui.rs:1949]: Fixed in 6ec8cb5: `total_wrapped_lines`/`get_max_detail_scroll` are now computed once per scroll event instead of once per stepped row, with the step applied as a single arithmetic op (factored the shared offset/bound computation into `feed_scroll_bounds` so it's not duplicated between keyboard and mouse paths). The O(1) `move_cursor_up/down` branch keeps its step loop since there's no expensive recomputation there. All 168 confirm_tui tests plus the full 1353-test suite pass unchanged.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/agent/confirm_tui.rs [hunk 2]
```diff
@@ -1730,6 +1791,105 @@ fn handle_key(dash: &Arc<RatatuiDashboard>, key: KeyEvent) {
...
+/// drive a redraw storm (issue #734 review).
+fn handle_mouse(dash: &Arc<RatatuiDashboard>, mouse: MouseEvent) -> bool {
+    let mut s = dash
+        .state
+        .lock()
+        .unwrap_or_else(std::sync::PoisonError::into_inner);
+
+    let modal_owns_focus = s.help_open
+        || s.stop_pending
+        || s.pending.is_some()
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/agent/confirm_tui.rs]: Good catch, fixed in 8a8e1db: `modal_owns_focus` now only treats `pending` as focus-owning when `detail_open` is false, matching `handle_key`'s precedence (its own `s.detail_open` arm is nested inside `s.pending.take()`). Added tests confirming wheel scroll now reaches the detail pane in that state without disturbing the pending prompt, and that a click still can't select a feed row (the feed isn't visible then). All 173 confirm_tui tests + full 1358-test suite pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/agent/confirm_tui.rs [hunk 3]
```diff
@@ -849,16 +896,20 @@ async fn renderer_loop(
...
         }
+        redraw = true;
         tokio::select! {
             _ = &mut shutdown => break,
             () = &mut notified => {}
             _ = tick.tick(), if active => {}
             ev = events.next() => {
                 match ev {
                     Some(Ok(Event::Key(key))) => handle_key(&dash, key),
+                    Some(Ok(Event::Mouse(mouse))) => redraw = handle_mouse(&dash, mouse),
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/agent/confirm_tui.rs]: Good catch — that race is real. Fixed in 9ad3211 by adding `biased;` to the `select!`. With the default random choice, an ignored mouse event could win a same-poll tie against a `notify_waiters()` call from another task, dropping that redraw since `Notify` stores no permit for the cancelled `notified` future. `biased;` makes `notified` always win ties against `events.next()` — cheap to do since every branch besides `events.next()` has a no-op body, and an event not chosen this poll just stays queued for the next one, so nothing is lost.

`renderer_loop` needs a live terminal so it isn't unit-testable (same as the rest of the file's event-loop plumbing), but the fix is a one-line, well-understood change to `select!`'s polling order. All 173 confirm_tui tests + full 1358-test suite still pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/agent/confirm_tui.rs [hunk 4]
```diff
@@ -1730,6 +1803,116 @@ fn handle_key(dash: &Arc<RatatuiDashboard>, key: KeyEvent) {
...
+    // review). `feedback_input`/`edit_input` still win unconditionally: they
+    // can only be set while `!detail_open` (`n`/`e` are only reachable from
+    // the non-detail-open branch of `handle_key`), so this never lets mouse
+    // input reach an active feedback/edit text buffer.
+    let modal_owns_focus = s.help_open
+        || s.stop_pending
+        || (s.pending.is_some() && !s.detail_open)
+        || s.feedback_input.is_some()
+        || s.edit_input.is_some()
+        || s.search.is_some();
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/agent/confirm_tui.rs]: Right, that's the same class of bug I fixed for `pending` but missed for `search`. Fixed in 319953c: `modal_owns_focus` now defers both `pending` and `search` to `detail_open`, matching `handle_key`'s routing (neither the pending-modal branch nor the "no modal open" branch ever consults `search` while `detail_open` is true). Added a test with `pending`, `search`, and `detail_open` all set simultaneously confirming wheel scroll reaches the detail pane without disturbing either. 175 confirm_tui tests + full 1360-test suite pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/agent/confirm_tui.rs [hunk 5]
```diff
@@ -849,16 +896,32 @@ async fn renderer_loop(
...
+            // the stream for the next one, so nothing is lost by picking
+            // `notified` first when both are ready.
+            biased;
             _ = &mut shutdown => break,
             () = &mut notified => {}
             _ = tick.tick(), if active => {}
             ev = events.next() => {
                 match ev {
                     Some(Ok(Event::Key(key))) => handle_key(&dash, key),
+                    Some(Ok(Event::Mouse(mouse))) => redraw = handle_mouse(&dash, mouse),
```
[the remark this answers is not in the document]
Comment (madmax983) [reply src/agent/confirm_tui.rs]: Confirmed, `biased` only closed the same-poll tie, not this narrower window. Fixed in 319953c with a structural fix rather than another timing patch: added `RedrawNotify`, a thin wrapper around `Notify` that bumps a monotonic version counter (`Release` on write, `Acquire` on read) on every `notify_waiters()` call — no changes needed at any of the ~60 existing call sites since the method name/signature are unchanged. `renderer_loop` now ORs `handle_mouse`'s result with `dash.notify.version() != last_drawn_version`, so it detects "something changed since my last draw" regardless of exactly when the other task's notify landed relative to `select!`'s polling, rather than relying on which branch happened to win. Added a unit test for the version counter itself (`renderer_loop` isn't otherwise testable — needs a live terminal). 175 confirm_tui tests + full 1360-test suite pass.

---
_Generated by [Claude Code](https://claude.ai/code)_
